### PR TITLE
feat!: bumped ESLint packages to latest and added recommended & Jest linting

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### Overview

Updates all our ESLint plugins to the latest and adds some around Airbnb rulesets and Jest.

Adds a couple new packages:

* https://www.npmjs.com/package/eslint-plugin-jest: The standard community Jest lint rule set
* https://www.npmjs.com/package/eslint-plugin-jest-react: New one we're creating (https://github.com/codecademy-engineering/Codecademy/pull/21529)

This will kick off a whole bunch of pairing opportunities for folks to work with me on enabling these new lint rules in the monolith and static-sites.

See equivalent PRs in consuming apps:
* https://github.com/codecademy-engineering/author/pull/1458
* https://github.com/codecademy-engineering/Codecademy/pull/21597 
* https://github.com/codecademy-engineering/static-sites/pull/538

### PR Checklist

- ~[ ] Related to designs:~
- [x] Related to JIRA ticket: [WEB-1260](https://codecademy.atlassian.net/browse/WEB-1260)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.4-alpha.30079d.0<br/>@codecademy/gamut-illustrations@0.4.10-alpha.30079d.0<br/>@codecademy/gamut-labs@7.13.5-alpha.30079d.0<br/>@codecademy/gamut-styles@7.6.4-alpha.30079d.0<br/>@codecademy/gamut-system@0.2.2-alpha.30079d.0<br/>@codecademy/gamut@19.11.6-alpha.30079d.0<br/>@codecademy/markdown-overrides@0.4.4-alpha.30079d.0<br/>@codecademy/styleguide@21.20.3-alpha.30079d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6010ae8993030b2deb615894--gamut-preview.netlify.app>https://6010ae8993030b2deb615894--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6010ae8993030b2deb615894>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6010d76228223f73d6833a8c--gamut-preview.netlify.app>https://6010d76228223f73d6833a8c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6010d76228223f73d6833a8c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.4-alpha.998fb0.0<br/>@codecademy/gamut-illustrations@0.4.10-alpha.998fb0.0<br/>@codecademy/gamut-labs@7.13.5-alpha.998fb0.0<br/>@codecademy/gamut-styles@7.6.4-alpha.998fb0.0<br/>@codecademy/gamut-system@0.2.2-alpha.998fb0.0<br/>@codecademy/gamut@19.11.6-alpha.998fb0.0<br/>@codecademy/macros@1.1.3-alpha.998fb0.0<br/>@codecademy/markdown-overrides@0.4.4-alpha.998fb0.0<br/>@codecademy/styleguide@21.20.3-alpha.998fb0.0<br/>@codecademy/webpack-config@5.2.2-alpha.998fb0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6010dc0593030b7a92615323--gamut-preview.netlify.app>https://6010dc0593030b7a92615323--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6010dc0593030b7a92615323>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.4-alpha.41cb0e.0<br/>@codecademy/gamut-illustrations@0.4.10-alpha.41cb0e.0<br/>@codecademy/gamut-labs@7.13.5-alpha.41cb0e.0<br/>@codecademy/gamut-styles@7.6.4-alpha.41cb0e.0<br/>@codecademy/gamut-system@0.2.2-alpha.41cb0e.0<br/>@codecademy/gamut@19.11.6-alpha.41cb0e.0<br/>@codecademy/macros@1.1.3-alpha.41cb0e.0<br/>@codecademy/markdown-overrides@0.4.4-alpha.41cb0e.0<br/>@codecademy/styleguide@21.20.3-alpha.41cb0e.0<br/>@codecademy/webpack-config@5.2.2-alpha.41cb0e.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.4-alpha.e5c55f.0<br/>@codecademy/gamut-illustrations@0.4.10-alpha.e5c55f.0<br/>@codecademy/gamut-labs@7.13.5-alpha.e5c55f.0<br/>@codecademy/gamut-styles@7.6.4-alpha.e5c55f.0<br/>@codecademy/gamut-system@0.2.2-alpha.e5c55f.0<br/>@codecademy/gamut@19.11.6-alpha.e5c55f.0<br/>@codecademy/macros@1.1.3-alpha.e5c55f.0<br/>@codecademy/markdown-overrides@0.4.4-alpha.e5c55f.0<br/>@codecademy/styleguide@21.20.3-alpha.e5c55f.0<br/>@codecademy/webpack-config@5.2.2-alpha.e5c55f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601181c42a7ccf89c5c012da--gamut-preview.netlify.app>https://601181c42a7ccf89c5c012da--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601181c42a7ccf89c5c012da>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6011a1b438b5ef780fa1112f--gamut-preview.netlify.app>https://6011a1b438b5ef780fa1112f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6011a1b438b5ef780fa1112f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.4-alpha.fb611d.0<br/>@codecademy/gamut-illustrations@0.4.10-alpha.fb611d.0<br/>@codecademy/gamut-labs@7.13.5-alpha.fb611d.0<br/>@codecademy/gamut-styles@7.6.4-alpha.fb611d.0<br/>@codecademy/gamut-system@0.2.2-alpha.fb611d.0<br/>@codecademy/gamut@19.11.6-alpha.fb611d.0<br/>@codecademy/macros@1.1.3-alpha.fb611d.0<br/>@codecademy/markdown-overrides@0.4.4-alpha.fb611d.0<br/>@codecademy/styleguide@21.20.3-alpha.fb611d.0<br/>@codecademy/webpack-config@5.2.2-alpha.fb611d.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.4-alpha.e4923c.0<br/>@codecademy/gamut-illustrations@0.4.10-alpha.e4923c.0<br/>@codecademy/gamut-labs@7.13.5-alpha.e4923c.0<br/>@codecademy/gamut-styles@7.6.4-alpha.e4923c.0<br/>@codecademy/gamut-system@0.2.2-alpha.e4923c.0<br/>@codecademy/gamut@19.11.6-alpha.e4923c.0<br/>@codecademy/macros@1.1.3-alpha.e4923c.0<br/>@codecademy/markdown-overrides@0.4.4-alpha.e4923c.0<br/>@codecademy/styleguide@21.20.3-alpha.e4923c.0<br/>@codecademy/webpack-config@5.2.2-alpha.e4923c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6011a60c4e99467813d7daf2--gamut-preview.netlify.app>https://6011a60c4e99467813d7daf2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6011a60c4e99467813d7daf2>Deploy Logs</a></p>

File: packages/eslint-config/index.js [hunk 1]
```diff
@@ -55,6 +59,10 @@ module.exports = {
         '@typescript-eslint/require-await': 'off',
         '@typescript-eslint/prefer-regexp-exec': 'off',
         '@typescript-eslint/unbound-method': 'off',
+        'import/no-named-as-default': 'off',
+        'no-restricted-globals': 'off',
+        'no-undef': 'off',
```
Comment (codecaaron) [inline packages/eslint-config/index.js:64]: Just curious was this turned off through an extension before?  Or is this net new?

File: packages/gamut/src/FlexBox/Item.tsx [hunk 2]
```diff
@@ -2,13 +2,12 @@ import React from 'react';
...
-
-  render() {
-    return <Container {...this.props} />;
-  }
-}
+export const Item: React.FC<ContainerProps> = (props) => {
+  return <Container {...props} />;
+};
+
+Item.displayName = 'Item';
```
Comment (codecaaron) [inline packages/gamut/src/FlexBox/Item.tsx:9]: Is this not getting covered by the babel plugin?  Just curious.

File: packages/gamut/src/FlexBox/Container.tsx [hunk 3]
```diff
@@ -70,40 +70,38 @@ export interface ContainerProps extends HTMLAttributes<HTMLDivElement> {
...
-    });
+  return (
+    <div {...propsToTransfer} className={classes}>
+      {props.children}
+    </div>
+  );
+};
 
-    const propsToTransfer = omit(this.props, internalProps);
+Container.displayName = 'Container';
```
Comment (codecaaron) [inline packages/gamut/src/FlexBox/Container.tsx:102]: Was this broken before?

File: packages/gamut/src/InputStepper/index.tsx [hunk 4]
```diff
@@ -82,7 +82,7 @@ export const InputStepper: React.FC<InputStepperProps> = ({
 
   const onChangeEvent = (event: React.ChangeEvent<HTMLInputElement>) => {
     const rawValue = event.target.value || `${min}`;
-    const normalizedValue = normalize(parseInt(rawValue));
+    const normalizedValue = normalize(parseInt(rawValue, 10));
```
Comment (codecaaron) [inline packages/gamut/src/InputStepper/index.tsx:85]: Yay radix enforcement

**Review by codecaaron — APPROVED** [review]
Nice!

Lots of great rules that make my heart sing!

File: packages/gamut/src/FlexBox/Item.tsx (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/FlexBox/Item.tsx:9]: Oh yeah we can remove these now.

File: packages/eslint-config/index.js (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/eslint-config/index.js:64]: I want to say it was turned off through extensions before, and now isn't. It was a little irksome playing around trying to find where they came from...

File: packages/eslint-config/index.js [hunk 5]
```diff
@@ -6,14 +6,17 @@ module.exports = {
     node: true,
   },
   extends: [
+    'eslint:recommended',
     'plugin:import/errors',
     'plugin:import/typescript',
     'plugin:jsx-a11y/strict',
     'plugin:react/recommended',
+    'plugin:react-hooks/recommended',
+    'airbnb',
```
Comment (jakemhiller) [inline packages/eslint-config/index.js:15]: Hello old friend

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.4-alpha.d0ce1d.0<br/>@codecademy/gamut-illustrations@0.4.10-alpha.d0ce1d.0<br/>@codecademy/gamut-labs@7.13.5-alpha.d0ce1d.0<br/>@codecademy/gamut-styles@7.6.4-alpha.d0ce1d.0<br/>@codecademy/gamut-system@0.2.2-alpha.d0ce1d.0<br/>@codecademy/gamut@19.11.6-alpha.d0ce1d.0<br/>@codecademy/macros@1.1.3-alpha.d0ce1d.0<br/>@codecademy/markdown-overrides@0.4.4-alpha.d0ce1d.0<br/>@codecademy/styleguide@21.20.3-alpha.d0ce1d.0<br/>@codecademy/webpack-config@5.2.2-alpha.d0ce1d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6011acaedc571c6dec5f8a0a--gamut-preview.netlify.app>https://6011acaedc571c6dec5f8a0a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6011acaedc571c6dec5f8a0a>Deploy Logs</a></p>

**Review by jakemhiller — CHANGES_REQUESTED** [review]
Would prefer to merge this when we have autofixing enabled on commit. Based on the changes in this PR it seems like there are numerous style-based lint rules that we normally don't block PRs on because things are fixed automatically.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.4-alpha.3b9d29.0<br/>@codecademy/gamut-illustrations@0.4.10-alpha.3b9d29.0<br/>@codecademy/gamut-labs@7.13.5-alpha.3b9d29.0<br/>@codecademy/gamut-styles@7.6.4-alpha.3b9d29.0<br/>@codecademy/gamut-system@0.2.2-alpha.3b9d29.0<br/>@codecademy/gamut@19.11.6-alpha.3b9d29.0<br/>@codecademy/macros@1.1.3-alpha.3b9d29.0<br/>@codecademy/markdown-overrides@0.4.4-alpha.3b9d29.0<br/>@codecademy/styleguide@21.20.3-alpha.3b9d29.0<br/>@codecademy/webpack-config@5.2.2-alpha.3b9d29.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.7-alpha.d9e545.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.d9e545.0<br/>@codecademy/gamut-labs@8.0.3-alpha.d9e545.0<br/>@codecademy/gamut-styles@7.7.1-alpha.d9e545.0<br/>@codecademy/gamut-system@0.2.2-alpha.d9e545.0<br/>@codecademy/gamut@20.0.6-alpha.d9e545.0<br/>@codecademy/macros@1.1.3-alpha.d9e545.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.d9e545.0<br/>@codecademy/styleguide@23.1.2-alpha.d9e545.0<br/>@codecademy/webpack-config@5.2.2-alpha.d9e545.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6018567f2827571194943cca--gamut-preview.netlify.app>https://6018567f2827571194943cca--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6018567f2827571194943cca>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.7-alpha.a8a6bf.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.a8a6bf.0<br/>@codecademy/gamut-labs@8.0.3-alpha.a8a6bf.0<br/>@codecademy/gamut-styles@7.7.1-alpha.a8a6bf.0<br/>@codecademy/gamut-system@0.2.2-alpha.a8a6bf.0<br/>@codecademy/gamut@20.0.6-alpha.a8a6bf.0<br/>@codecademy/macros@1.1.3-alpha.a8a6bf.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.a8a6bf.0<br/>@codecademy/styleguide@23.1.2-alpha.a8a6bf.0<br/>@codecademy/webpack-config@5.2.2-alpha.a8a6bf.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60185c2cd1b0941c4ec33460--gamut-preview.netlify.app>https://60185c2cd1b0941c4ec33460--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60185c2cd1b0941c4ec33460>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.7-alpha.c96de7.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.c96de7.0<br/>@codecademy/gamut-labs@8.0.3-alpha.c96de7.0<br/>@codecademy/gamut-styles@7.7.1-alpha.c96de7.0<br/>@codecademy/gamut-system@0.2.2-alpha.c96de7.0<br/>@codecademy/gamut@20.0.6-alpha.c96de7.0<br/>@codecademy/macros@1.1.3-alpha.c96de7.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.c96de7.0<br/>@codecademy/styleguide@23.1.2-alpha.c96de7.0<br/>@codecademy/webpack-config@5.2.2-alpha.c96de7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601dcab23966b268d406bc81--gamut-preview.netlify.app>https://601dcab23966b268d406bc81--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601dcab23966b268d406bc81>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.2.0-alpha.f966ff.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.f966ff.0<br/>@codecademy/gamut-labs@8.0.3-alpha.f966ff.0<br/>@codecademy/gamut-styles@7.7.1-alpha.f966ff.0<br/>@codecademy/gamut-system@0.2.2-alpha.f966ff.0<br/>@codecademy/gamut@20.0.6-alpha.f966ff.0<br/>@codecademy/macros@1.1.3-alpha.f966ff.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.f966ff.0<br/>@codecademy/styleguide@23.1.2-alpha.f966ff.0<br/>@codecademy/webpack-config@5.2.2-alpha.f966ff.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60218439b3a45c008deea67f--gamut-preview.netlify.app>https://60218439b3a45c008deea67f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60218439b3a45c008deea67f>Deploy Logs</a></p>

**Review by jakemhiller — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.2.0-alpha.9686fe.0<br/>@codecademy/gamut-illustrations@0.4.30-alpha.9686fe.0<br/>@codecademy/gamut-labs@9.2.5-alpha.9686fe.0<br/>@codecademy/gamut-styles@7.7.4-alpha.9686fe.0<br/>@codecademy/gamut-system@0.3.1-alpha.9686fe.0<br/>@codecademy/gamut-tests@1.1.2-alpha.9686fe.0<br/>@codecademy/gamut@21.3.3-alpha.9686fe.0<br/>@codecademy/macros@1.1.3-alpha.9686fe.0<br/>@codecademy/markdown-overrides@0.4.8-alpha.9686fe.0<br/>@codecademy/styleguide@25.1.6-alpha.9686fe.0<br/>@codecademy/webpack-config@5.2.4-alpha.9686fe.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60218ddcfdd9fd0a44c31551--gamut-preview.netlify.app>https://60218ddcfdd9fd0a44c31551--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60218ddcfdd9fd0a44c31551>Deploy Logs</a></p>

**Review by hmafzal — APPROVED** [review]
LGTM! +1 to Jake

**Review by MattSchiller — APPROVED** [review]
Awesome & all about this =)

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.2.0-alpha.1242cd.0<br/>@codecademy/gamut-illustrations@0.4.36-alpha.1242cd.0<br/>@codecademy/gamut-labs@9.3.8-alpha.1242cd.0<br/>@codecademy/gamut-styles@7.7.4-alpha.1242cd.0<br/>@codecademy/gamut-system@0.3.1-alpha.1242cd.0<br/>@codecademy/gamut-tests@1.1.3-alpha.1242cd.0<br/>@codecademy/gamut@21.3.9-alpha.1242cd.0<br/>@codecademy/macros@1.1.3-alpha.1242cd.0<br/>@codecademy/markdown-overrides@0.4.8-alpha.1242cd.0<br/>@codecademy/styleguide@25.2.8-alpha.1242cd.0<br/>@codecademy/webpack-config@5.2.4-alpha.1242cd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6026d94e11829739d289572c--gamut-preview.netlify.app>https://6026d94e11829739d289572c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6026d94e11829739d289572c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>babel-preset-codecademy@3.1.1-alpha.efb7d6.0<br/>@codecademy/eslint-config@3.2.0-alpha.efb7d6.0<br/>@codecademy/gamut-illustrations@0.4.36-alpha.efb7d6.0<br/>@codecademy/gamut-labs@9.3.8-alpha.efb7d6.0<br/>@codecademy/gamut-styles@7.7.4-alpha.efb7d6.0<br/>@codecademy/gamut-system@0.3.1-alpha.efb7d6.0<br/>@codecademy/gamut-tests@1.1.3-alpha.efb7d6.0<br/>@codecademy/gamut@21.3.9-alpha.efb7d6.0<br/>@codecademy/macros@1.1.3-alpha.efb7d6.0<br/>@codecademy/markdown-overrides@0.4.8-alpha.efb7d6.0<br/>@codecademy/styleguide@25.2.8-alpha.efb7d6.0<br/>@codecademy/webpack-config@5.2.4-alpha.efb7d6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6026dae67c6b9f312d923658--gamut-preview.netlify.app>https://6026dae67c6b9f312d923658--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6026dae67c6b9f312d923658>Deploy Logs</a></p>

**Pull request merged**